### PR TITLE
[V2] Change exchange-wrapper pipe to pipe algorithm directly instead of fetching on wrapper end

### PR DESCRIPTION
**Pull request by andrlime**

### CODE REVIEW

File: exchange/src/wrapper/messaging/comms.cpp [hunk 1]
```diff
@@ -101,15 +103,20 @@ ExchangeProxy::publish_message(const std::string& message)
     lock.unlock();
 }
 
-std::variant<start_time, tick_update>
-ExchangeProxy::consume_message()
+algorithm_t
+ExchangeProxy::consume_algorithm() {
+    return consume_message<algorithm_t>();
+}
```
Comment (glasss13) [inline exchange/src/wrapper/messaging/comms.cpp:109]: Nit but I feel like we probably don't need this function.

Comment (andrlime) [reply exchange/src/wrapper/messaging/comms.cpp:109]: My reasoning was that `consume_message()` was private for a reason, so we should keep it that way?

File: exchange/src/shared/messages_exchange_to_wrapper.hpp [hunk 2]
```diff
@@ -115,3 +124,11 @@ struct glz::meta<nutc::messages::start_time> {
     static constexpr auto value = // NOLINT
         object(&t::start_time_ns);
 };
+
+/// \cond
+template <>
+struct glz::meta<nutc::messages::algorithm_content> {
```
Comment (stevenewald) [inline exchange/src/shared/messages_exchange_to_wrapper.hpp:130]: we should clean this and start_message up

this is good for now (and improvement is outside the scope of this PR)

File: exchange/src/wrapper/messaging/comms.cpp (hunk 1 above)
Comment (stevenewald) [reply exchange/src/wrapper/messaging/comms.cpp:109]: I think this is a code smell. I'm okay with it for now

**Review by stevenewald — APPROVED** [review]
Great work, LGTM

**Comment by stevenewald** [comment]
Pls squash then feel free to merge

**Pull request merged**